src/test/java/com/google/api/codegen/testdata/csharp_gapic_resourcenames_library.baseline

```diff
@@ -0,0 +1,719 @@
+        {
+            // TODO: Improve exceptions
+            if (array == null) throw new ArgumentNullException();
+            if (arrayIndex < 0) throw new ArgumentOutOfRangeException();
+            if (Count > array.Length - arrayIndex) throw new ArgumentException();
+            for (int i = 0; i < Count; i++)
+            {
+                array[arrayIndex + i] = _stringToName(_underlyingList[i]);
+            }
+        }
+        public IEnumerator<TResource> GetEnumerator() => _underlyingList.Select(_stringToName).GetEnumerator();
+        System.Collections.IEnumerator System.Collections.IEnumerable.GetEnumerator() => GetEnumerator();
+        public int IndexOf(TResource value) => _underlyingList.IndexOf(_nameToString(value));
+        public void Insert(int index, TResource item) => _underlyingList.Insert(index, _nameToString(item));
+        public bool Remove(TResource item) => _underlyingList.Remove(_nameToString(item));
+        public void RemoveAt(int index) => _underlyingList.RemoveAt(index);
+    }
+
+    /// <summary>
+    /// Resource name for a shelf.
+    /// </summary>
+    public sealed partial class ShelfName : ISingleResourceName
+    {
+        private static readonly PathTemplate s_template = new PathTemplate("shelves/{shelf_id}");
+
+        /// <summary>
+        /// Parses a shelf resource name in string form into a <see cref="ShelfName"/>.
+        /// </summary>
+        /// <param name="shelfName">The shelf resource name in string form. Must not be <c>null</c>.</param>
+        public static ShelfName Parse(string shelfName)
+        {
+            GaxPreconditions.CheckNotNull(shelfName, nameof(shelfName));
+            ResourceName resourceName = s_template.ParseName(shelfName);
+            return new ShelfName(resourceName[0]);
+        }
+
+        public static bool TryParse(string shelfName, out ShelfName result)
+        {
+            GaxPreconditions.CheckNotNull(shelfName, nameof(shelfName));
+            ResourceName resourceName;
+            if (s_template.TryParseName(shelfName, out resourceName))
+            {
+                result = new ShelfName(resourceName[0]);
+                return true;
+            }
+            else
+            {
+                result = null;
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// Constructs a shelf name from its component parts.
+        /// </summary>
+        /// <param name="shelfIdId">The shelfId ID. Must not be <c>null</c>.</param>
+        public ShelfName(string shelfIdId)
+        {
+            ShelfIdId = GaxPreconditions.CheckNotNull(shelfIdId, nameof(shelfIdId));
+        }
+
+        /// <summary>
+        /// The shelfId ID. Never <c>null</c>.
+        /// </summary>
+        public string ShelfIdId { get; }
+
+        /// <inheritdoc />
+        public override string ToString() => s_template.Expand(ShelfIdId);
+    }
+
+    /// <summary>
+    /// Resource name for a archived_book.
```

Comment:
a archived_book?

- A/an (annoying to fix; can we reword to avoid it being a problem?)
- Is archived_book meant to refer to something the user would know about? Maybe we need a configurable documentation name